tests/taproot_multi_maker.rs

```diff
@@ -0,0 +1,197 @@
+    taproot_makers
+        .iter()
+        .for_each(|maker| maker.shutdown.store(true, Relaxed));
+
+    taproot_maker_threads
+        .into_iter()
+        .for_each(|thread| thread.join().unwrap());
+
+    // Sync wallets and verify results
+    taproot_taker.get_wallet_mut().sync_and_save().unwrap();
+
+    // Mine a block to confirm the sweep transactions
+    generate_blocks(bitcoind, 1);
+
+    // Synchronize each taproot maker's wallet multiple times to ensure all UTXOs are discovered
+    for maker in taproot_makers.iter() {
+        let mut wallet = maker.wallet().write().unwrap();
+        wallet.sync_and_save().unwrap();
+    }
+    let taker_wallet = taproot_taker.get_wallet();
+    let taker_balances = taker_wallet.get_balances().unwrap();
+
+    // Use spendable balance (regular + swap) since swept coins from V2 swaps
+    // are tracked as SweptCoinV2 and appear in swap balance
+    let taker_total_after = taker_balances.spendable;
+    assert!(
+        taker_total_after.to_sat() == 14860649, // Multi maker case (less spendable balance due to higher no. of makers)
+        "Taproot Taker spendable balance check. Original: {}, After: {}",
+        taproot_taker_original_balance,
+        taker_total_after
+    );
+
+    let balance_diff = taproot_taker_original_balance - taker_total_after;
+    assert!(
+        balance_diff.to_sat() == 139351, // amount spended as fee (multi maker case,more no. of maker so more fee)
+        "Taproot taker spent on fees check. Original: {}, After: {}, Max expected fees: {}",
+        taproot_taker_original_balance,
+        taker_total_after,
+        balance_diff
+    );
+
+    info!(
+        "Taproot Taker balance verification passed. Original spendable: {}, After spendable: {} (fees paid: {})",
+        taproot_taker_original_balance,
+        taker_total_after,
+        balance_diff
+    );
+
+    // Verify makers earned fees
+    for (i, (maker, original_spendable)) in taproot_makers
+        .iter()
+        .zip(actual_maker_spendable_balances)
+        .enumerate()
+    {
+        let wallet = maker.wallet().read().unwrap();
+        let balances = wallet.get_balances().unwrap();
+
+        info!(
+            "Taproot Maker {} final balances - Regular: {}, Swap: {}, Contract: {}, Fidelity: {}, Spendable: {}",
+            i, balances.regular, balances.swap, balances.contract, balances.fidelity, balances.spendable
+        );
+
+        // In multi maker it's hard to track the absolute spendable balance of a
+        // particular maker, as each time their order can be different leading to
+        // differnt spendable balances.
+        let max_fees = Amount::from_sat(100000); // Maximum expected mining fees
+        let expected_minimum = original_spendable - max_fees;
+        assert!(
+            balances.spendable >= expected_minimum,
+            "Taproot Maker {} balance should not change significantly. Original spendable: {}, Current spendable: {}",
```

Comment:
This hugely off. Fees are about 30% of the above max_fee value. We should just assert individual makers' fees.